tests/ops/models/privacy_request/test_privacy_request.py

```diff
@@ -1132,3 +1144,139 @@ def test_persist_custom_identities(self, db, privacy_request):
             customer_id=LabeledIdentity(label="Custom ID", value=123),
             account_id=LabeledIdentity(label="Account ID", value="456"),
         )
+
+
+def test_retrieve_attachments_from_privacy_request(
+    s3_client, db, attachment_data, privacy_request, monkeypatch
+):
+    # Create Attachments
+
+    def mock_get_s3_client(auth_method, storage_secrets):
+        return s3_client
+
+    monkeypatch.setattr(
+        "fides.api.service.storage.s3.get_s3_client", mock_get_s3_client
+    )
+
+    attachment1 = Attachment.create_and_upload(
+        db=db,
+        data=attachment_data,
+        attachment_file=b"contents of test file 1",
+    )
```

Comment:
If we move this away from the `Attachment` model and to the `StorageService`, we could test this with
```
mock_s3_storage_provider = S3StorageProvider(s3_client=mock_s3_client)
storage_service = StorageService(storage_provider=mock_s3_storage_provider)
storage_service.upload_attachment(attachment_data, attachment_file)
```